### ISSUE TITLE
Reaction list condensation

### ISSUE THREAD

**Issue by Stevel-Knievel** [angarg12/incremental_table_elements#155]
The as one unlocks more possible reactions, the list becomes far more complex, and it's nigh-impossible for some elements to find the exact reaction you want in the list, e.g. Carbon. It would be amazing if a filterable/searchable list function can be added to the game. Also, reactions with an inverse I feel should be listed as one option, with a "reverse" button, like the one in ITE.

**Comment by angarg12** [comment]
What about a filter like the one in the dashboard? Probably with two fields, one for reactant and one for product.

**Comment by Stevel-Knievel** [comment]
That would be nice, I think that's a good place to start. In my opinion, I think the end product should be one that has a web of reactions, starting with the raw element, and branching out to other possible reactions. Also, in my opinion, I think that reactions that you've unlocked but can't do because of lack of resources should be hidden. 

(In my mind's eye, I'm thinking of something like this:

![image](https://user-images.githubusercontent.com/33704300/33348417-dd172202-d442-11e7-9cbc-39f99f9fad83.png)

where reactions are invisible until you have all component elements unlocked (and in use, or in high quantity), then are greyed out until one has the ability to actually synthesize it {like if I'm going for something like H2SO4, which requires the SO4(-2) ion and a diatomic hydrogen molecule, it wouldn't fully brighten as a possibility until I had all of the components created and ready.}

P.S. I'm sorry this is a bit rushed, I'm typing this on a new keyboard and I have a bus to catch

**Comment by primeprover** [comment]
If you hide unlocked reactions that you don't have resources for then you might never find them. A possible option maybe.

**Comment by Stevel-Knievel** [comment]
That is true. Idk, I'm looking at it from having played Realm Grinder, where Researches are hidden for the most part until you complete the unlock conditions, until then, all you have is a vague percentage. Idk, over Christmas Break I'm planning on devoting time to doing the math for heat energy and reaction energy, phase change, and such. (Preferably I can make it as a program, but that's a stretch for me) That way it'll be easier to work with electronegativity, ionization, reaction enthalpy, etc. Also, if I can finish that, I plan on putting time into trying to work out catalysts. But I digress, what I'm trying to say is that once I work out all the reaction values and such, then it'll be easier to seed hints for those who want to find them. 
For example, giving the reaction enthalpy for a person to work back from to find the exact reactants, or giving an endo/exothermic symbol, if temperature management becomes a thing.

(Another sidenote, I'd love for there to be a mechanic for certain weird reactions like pyrolysis and acid-base chemistry and such) 

Third side note, because ADHD thinking, I'd totally love for there to be actual chemistry analysis tools as parts of programmed "quests" that give access to really big molecules or certain specific processes, like "using" NMR to determine what a molecule is to unlock it (by filling in a model or something like that with the help of an information guide and tutorial), or using stepdown cooling for creating liquid versions of H2, O2, etc, assuming that thermodynamics, and thus state-change physics, are applied to the game, with phase playing a role in reactions and achievements.
